### PR TITLE
SLR - SL-142

### PR DESCRIPTION
**Pull request by lucatume**
This updates the code, where required, to support the customization features required by SLR in the [SL-142 ticket](https://stellarwp.atlassian.net/browse/SL-142).

CleanShot 2024-09-03 at 17 55 04

### CODE REVIEW

File: tests/wpunit/Admin/Fields/FieldTest.php [hunk 1]
```diff
@@ -213,4 +245,35 @@ public function it_should_handle_empty_label( $resource ) {
...
+		Register::service(
+			$slug,
+			'Service OAuth With License Key Field 1',
+			'1.0.0',
+			$this->get_base() . '/service-oauth-with-license-key-field-1.php',
+			Uplink::class,
+			Uplink::class,
+			Resource::OAUTH_REQUIRES_LICENSE_KEY
+		);
+		add_filter( 'stellarwp/uplink/test/auth/can_auth', '__return_true' );
```
Comment (dpanta94) [inline tests/wpunit/Admin/Fields/FieldTest.php:265]: This is getting cleaned up after the test ?

**Review by dpanta94 — CHANGES_REQUESTED** [review]
Looks good! Just an apply filters i'd like us we implement and a question!

File: src/Uplink/Admin/Fields/Field.php [hunk 2]
```diff
@@ -235,17 +235,23 @@ public function render( array $args = [] ): void {
...
+		$args = [
+			'field' => $this,
+			'group' => $this->group->get_name( $this->get_slug() ),
+		];
+
 		if ( $this->resource->is_using_oauth() ) {
 			ob_start();
+
+			if ( $this->resource->oauth_requires_license_key() ) {
+				echo $this->view->render( self::VIEW, $args );
```
[the remark this answers is not in the document]
Comment (lucatume) [reply src/Uplink/Admin/Fields/Field.php:247]: Good idea.

File: tests/wpunit/Admin/Fields/FieldTest.php (hunk 1 above)
Comment (lucatume) [reply tests/wpunit/Admin/Fields/FieldTest.php:265]: Yes, like all filters they are being reset to the original between tests.

**Review by dpanta94 — APPROVED** [review]

**Pull request merged**